dali/operators/reader/loader/tar_utils.cc

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/tar_utils.h"
+#include <libtar.h>
+#include <algorithm>
+#include <cstdarg>
+#include <cstdlib>
+#include <list>
+#include <string>
+#include <tuple>
+#include <utility>
+#include "dali/core/error_handling.h"
+
+namespace dali {
+namespace detail {
+
+namespace {
+
+constexpr uint64_t kBlockSize = 512;
+constexpr uint64_t kEmptyEofBlocks = 2;
+constexpr uint64_t kTarArchiveBufferInitSize = 1;
+
+std::mutex instances_mutex;
+std::list<std::vector<TarArchive*>> instances_registry = {
+    std::vector<TarArchive*>(kTarArchiveBufferInitSize)};
+TarArchive** instances = instances_registry.back().data();
+
+}  // namespace
+
+int Register(TarArchive* archive) {
```

Comment:
Move this to the anonymous namespace or add static / inline.

Reply:
alright